python/src/iceberg/expression/literals.py

```diff
@@ -0,0 +1,742 @@
+
+    def to(self, type_var):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        if id(self) == id(other):
+            return True
+        elif other is None or not isinstance(other, BaseLiteral):
+            return False
+
+        return self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        return self._repr_string
+
+    def __str__(self):
+        return str(self.value)
+
+    def to_byte_buffer(self):
+        if self.byte_buffer is None:
+            # TODO need byte buffer conversion
+            raise NotImplementedError()
+
+        return self.byte_buffer
+
+
+class ComparableLiteral(BaseLiteral):
+    def __init__(self, repr_string: str, value):
+        super(ComparableLiteral, self).__init__(repr_string, value)
+
+    def to(self, type_var):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __lt__(self, other):
+        if self.value is None:
+            return True
+
+        if other is None or other.value is None:
+            return False
+
+        return self.value < other.value
+
+    def __gt__(self, other):
+        if self.value is None:
+            return False
+
+        if other is None or other.value is None:
+            return True
+
+        return self.value > other.value
+
+    def __le__(self, other):
+        if self.value is None:
+            return True
+
+        if other is None or other.value is None:
+            return False
+
+        return self.value <= other.value
+
+    def __ge__(self, other):
+        if self.value is None:
+            return False
+
+        if other is None or other.value is None:
+            return True
+
+        return self.value >= other.value
+
+
+class AboveMax(Literal, Singleton):
+    _instance = None
+
+    def __init__(self):
+        self._initialized = True
+        super(AboveMax, self).__init__()
+
+    def value(self):
+        raise RuntimeError("AboveMax has no value")
+
+    def to(self, type_var):
+        raise RuntimeError("Cannot change the type of AboveMax")
+
+    def __str__(self):
+        return "aboveMax"
+
+
+class BelowMin(Literal, Singleton):
+    _instance = None
+
+    def __init__(self):
+        self._initialized = True
+        super(BelowMin, self).__init__()
+
+    def value(self):
+        raise RuntimeError("BelowMin has no value")
+
+    def to(self, type_var):
+        raise RuntimeError("Cannot change the type of BelowMin")
+
+    def __str__(self):
+        return "belowMin"
+
+
+class BooleanLiteral(ComparableLiteral):
+    def __init__(self, value):
+        super(BooleanLiteral, self).__init__(f"BooleanLiteral({value})", value)
+
+    @singledispatchmethod
+    def to(self, type_var):
+        return None
+
+    @to.register(BooleanType)
+    def boolean_literal_to(self, type_var):
```

Comment:
Just for consistency I would make name this `_` as well.

Reply:
yeah good catch, corrected it!